distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Text_Sub_Range.enso

```diff
@@ -0,0 +1,123 @@
+from Standard.Base import all
+from Standard.Base.Data.Text.Extensions import Index_Out_Of_Bounds_Error
+
+polyglot java import com.ibm.icu.text.BreakIterator
+polyglot java import org.enso.base.Text_Utils
+
+## Type defining a substring of a Text
+type Text_Sub_Range
+    ## Select the first `count` characters.
+       Select an empty string if `count` is less than or equal to 0.
+       Select the entire string if `count` is greater than the length of the input.
+    type First (count : Integer = 1)
+
+    ## Select the last `count` characters.
+       Select an empty string if `count` is less than or equal to 0.
+       Select the entire string if `count` is greater than the length of the input.
+    type Last (count : Integer = 1)
+
+    ## Select characters until the first instance of `delimiter`.
+       Select an empty string if `delimiter` is empty.
+       Select the entire string if the input does not contain `delimiter`.
+    type Before (delimiter : Text)
+
+    ## Select characters until the last instance of `delimiter`.
+       Select an empty string if `delimiter` is empty.
+       Select the entire string if the input does not contain `delimiter`.
+    type Before_Last (delimiter : Text)
+
+    ## Select characters after the first instance of `delimiter`.
+       Select an empty string if the input does not contain `delimiter`.
+    type After (delimiter : Text)
+
+    ## Select characters after the last instance of `delimiter`.
+       Select an empty string if the input does not contain `delimiter`.
+    type After_Last (delimiter : Text)
+
+    ## Select characters while the predicate returns `True`.
+    type While (predicate : (Text -> Boolean))
+
+    ## PRIVATE
+       Evaluates the Text_Sub_Range returning the underlying char array indices
```

Comment:
```suggestion
       Finds code-point indices corresponding to the part of the input matching the `Text_Sub_Range`.
```

I would consider rephrasing this as I don't think it really explains what it does. Evaluates is a very generic term.